### PR TITLE
Remove `flutter.yaml` handling logic (#656).

### PR DESCRIPTION
**Pull request by pq**
* remove checks for `flutter.yaml`
* remove editor notifications for `flutter.yaml` files
* update package explorer group to enable for flutter pubspecs

Fixes: #656.

Note that we've got a pile-up of actions suggested for `pubspec`s.  

screen shot 2017-01-24 at 1 28 21 pm

See: #660 for a proposal to fix this in the Dart Plugin.  Moreover, if the Dart editor notification for `pubspec`s gets suppressed we can recontribute `FlutterYamlNotificationProvider.java` (but keyed off Flutter pubspecs).

@devoncarew

### CODE REVIEW

File: src/io/flutter/FlutterUtils.java [hunk 1]
```diff
@@ -30,10 +32,15 @@ public static void invokeAndWait(@NotNull Runnable runnable) throws ProcessCance
 
   @SuppressWarnings("BooleanMethodIsAlwaysInverted")
   public static boolean isFlutteryFile(@NotNull VirtualFile file) {
-    final String fileName = file.getName();
-    return file.getFileType() == DartFileType.INSTANCE ||
-           fileName.equals(FlutterConstants.FLUTTER_YAML) ||
-           fileName.equals(FlutterConstants.PUBSPEC_YAML);
+    return isDartFile(file) || isPubspecFile(file);
```
Comment (devoncarew) [inline src/io/flutter/FlutterUtils.java]: We _may_ want to keep `flutter.yaml` in the check for a release or two (to identify the project as a flutter project), in order to give people a few weeks to migrate.

File: src/io/flutter/sdk/FlutterSdkUtil.java [hunk 2]
```diff
@@ -186,7 +186,6 @@ public static String getErrorMessageIfWrongSdkRootPath(final @NotNull String sdk
 
   public static boolean isFlutterProjectDir(@Nullable VirtualFile dir) {
     if (dir == null || !dir.isDirectory()) return false;
-    if (dir.findChild(FlutterConstants.FLUTTER_YAML) != null) return true;
```
Comment (devoncarew) [inline src/io/flutter/sdk/FlutterSdkUtil.java]: Same here re: removing the check; we may want to keep but remove after a certain time.

**Comment by devoncarew** [comment]
+1 to removing the UI and moving the contribution points.

We'll probably want a brief grace period for still identifying projects with flutter.yaml files as flutter projects.

lgtm

**Comment by pq** [comment]
> We'll probably want a brief grace period for still identifying projects with flutter.yaml files as flutter projects.

On thinking about it, this logic was always a little suspect --- isn't it true that all flutter projects need to have a pubspec dependency as well?  IOW: I think this check alone was never sufficient and once we started parsing pubspecs was already safe to remove.  What do you think?

File: src/io/flutter/FlutterUtils.java (hunk 1 above)
Comment (pq) [inline src/io/flutter/FlutterUtils.java]: See comment below.

File: src/io/flutter/sdk/FlutterSdkUtil.java (hunk 2 above)
Comment (pq) [inline src/io/flutter/sdk/FlutterSdkUtil.java]: And here too.  Net-net: I think the check was always fishy once we started reading pubspecs for Flutter dependencies.  Anyway, see my hypothesis below.

**Comment by devoncarew** [comment]
> isn't it true that all flutter projects need to have a pubspec dependency as well?

Thinking about it, that sounds likely. I think I'm still rooted in a world where we used relative paths to reference flutter.

**Pull request merged**